plugins/telemetry-otel/src/main/java/org/opensearch/telemetry/tracing/exporter/OTelSpanExporterFactory.java

```diff
@@ -45,21 +45,41 @@ private OTelSpanExporterFactory() {
      */
     public static SpanExporter create(Settings settings) {
         Class<SpanExporter> spanExporterProviderClass = OTelTelemetrySettings.OTEL_TRACER_SPAN_EXPORTER_CLASS_SETTING.get(settings);
-        SpanExporter spanExporter = instantiateSpanExporter(spanExporterProviderClass);
+        SpanExporter spanExporter = instantiateSpanExporter(spanExporterProviderClass, settings);
         logger.info("Successfully instantiated the SpanExporter class {}", spanExporterProviderClass);
         return spanExporter;
     }
 
-    private static SpanExporter instantiateSpanExporter(Class<SpanExporter> spanExporterProviderClass) {
+    private static SpanExporter instantiateSpanExporter(Class<SpanExporter> spanExporterProviderClass, Settings settings) {
         try {
             // Check we ourselves are not being called by unprivileged code.
             SpecialPermission.check();
             return AccessController.doPrivileged((PrivilegedExceptionAction<SpanExporter>) () -> {
                 try {
-                    return (SpanExporter) MethodHandles.publicLookup()
-                        .findStatic(spanExporterProviderClass, "create", MethodType.methodType(spanExporterProviderClass))
-                        .asType(MethodType.methodType(SpanExporter.class))
-                        .invokeExact();
+                    String methodName = "create";
+                    boolean createMethodWithSettingsParamExists = true;
+                    try {
+                        // Attempt to get the method with the specified name and parameter types
+                        spanExporterProviderClass.getMethod(methodName, Settings.class);
+                    } catch (NoSuchMethodException e) {
+                        createMethodWithSettingsParamExists = false;
```

Comment:
Better check the parameter type and don't decide on the exception. Sample code here - https://github.com/opensearch-project/OpenSearch/blob/d375e4cfea1a8e4c58dd1409365a67cf49366780/modules/lang-painless/src/main/java/org/opensearch/painless/ScriptClassInfo.java#L132